### PR TITLE
feat: Add UI changes in cluster group env

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Added highlight on input selection & text UI change in cluster group env in charts & cdpipeline

Fixes #[AB1873](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1873)

Screenshot 2023-02-01 at 4 21 37 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All testing is done manually by changing input & verifying the highlighted input selection in cdDetails react select & chart env select option

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/css/base.scss [hunk 1]
```diff
@@ -2517,3 +2517,11 @@ textarea::placeholder {
         }
     }
 }
+
+.dc__high-light-text {
```
Comment (shivani170) [inline src/css/base.scss:2521]: Any reason for  high-light and not highlight

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 2]
```diff
@@ -77,15 +82,31 @@ export const ChartEnvironmentSelector = ({
...
-                    Option: selectOption,
-                    GroupHeading
+                    SingleValue: singleOption,
+                    GroupHeading,
                 }}
                 classNamePrefix="values-environment-select"
                 placeholder="Select Environment"
                 value={selectedEnvironment}
-                styles={getCommonSelectStyle()}
+                styles={{
```
Comment (shivani170) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:91]: can we extract out styles in another component

File: src/css/base.scss (hunk 1 above)
Comment (milindk8) [reply src/css/base.scss]: done

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx (hunk 2 above)
Comment (milindk8) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx]: done

File: src/components/v2/common/ReactSelect.utils.tsx [hunk 3]
```diff
@@ -214,13 +214,32 @@ export function GroupHeading(props) {
 
 export function EnvFormatOptions(props) {
     const { data, environmentfieldName } = props
-    props.selectProps.styles.option = getCustomOptionSelectionStyle()
+    return <components.SingleValue {...props}>{data[environmentfieldName]}</components.SingleValue>
+}
+
+export function formatHighlightedText(option, { inputValue }, environmentfieldName) {
```
Comment (shivani170) [inline src/components/v2/common/ReactSelect.utils.tsx:220]: Please add type here

File: src/css/base.scss [hunk 4]
```diff
@@ -2517,3 +2517,11 @@ textarea::placeholder {
         }
     }
 }
+
+.dc__highlight-text {
+    mark {
+        padding: 0;
+        background-color: #ffd597 !important;
```
Comment (sohel-rp) [inline src/css/base.scss:2524]: Use Y300

File: src/components/cdPipeline/CDPipeline.tsx [hunk 5]
```diff
@@ -1085,18 +1085,22 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
         }
     }
 
-    selectOption = (props) => {
+    singleOption = (props) => {
         return <EnvFormatOptions {...props} environmentfieldName="name" />
     }
 
+    handleFormatHighlightedText = (opt, { inputValue }) => {
+        return formatHighlightedText(opt, { inputValue }, 'name')
```
Comment (sohel-rp) [inline src/components/cdPipeline/CDPipeline.tsx:1093]: Directly pass obj that contains inputValue

File: src/components/v2/common/ReactSelect.utils.tsx [hunk 6]
```diff
@@ -214,13 +214,32 @@ export function GroupHeading(props) {
...
+                    ),
+                }}
+            />
+            {option.clusterName && option.namespace && (
+                <small
+                    className="cn-6"
+                    dangerouslySetInnerHTML={{
+                        __html: (option.clusterName + '/' + option.namespace).replace(
+                            new RegExp(inputValue, 'gi'),
+                            (highlighted) => `<mark>${highlighted}</mark>`,
```
Comment (sohel-rp) [inline src/components/v2/common/ReactSelect.utils.tsx:238]: Extract this to be re-used

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 7]
```diff
@@ -51,10 +50,14 @@ export const ChartEnvironmentSelector = ({
     invalidaEnvironment,
 }: ChartEnvironmentSelectorType): JSX.Element => {
 
-    const selectOption = (props) => {
+    const singleOption = (props) => {
         return <EnvFormatOptions {...props} environmentfieldName="label" />
     }
 
+    const handleFormatHighlightedText = (opt, { inputValue }) => {
+        return formatHighlightedText(opt, { inputValue }, 'label')
```
Comment (sohel-rp) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:58]: same here

**Review by shivani170 — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**